eureka-client/src/main/java/com/netflix/appinfo/InstanceInfo.java

```diff
@@ -155,23 +159,23 @@ public InstanceInfo(
             @JsonProperty("lastDirtyTimestamp") Long lastDirtyTimestamp,
             @JsonProperty("actionType") ActionType actionType,
             @JsonProperty("asgName") String asgName) {
-        this.appName = appName;
-        this.appGroupName = appGroupName;
-        this.ipAddr = ipAddr;
-        this.sid = sid;
+        this.appName = appName!=null ? appName.intern():null;
+        this.appGroupName = appGroupName!=null? appGroupName.intern():null;
+        this.ipAddr = ipAddr!=null?ipAddr.intern():null;
+        this.sid = sid!=null?sid.intern():null;
         this.port = port;
         this.isUnsecurePortEnabled = portEnabled;
         this.securePort = securePort;
         this.isSecurePortEnabled = securePortEnabled;
-        this.homePageUrl = homePageUrl;
-        this.statusPageUrl = statusPageUrl;
-        this.healthCheckUrl = healthCheckUrl;
-        this.secureHealthCheckUrl = secureHealthCheckUrl;
-        this.vipAddress = vipAddress;
-        this.secureVipAddress = secureVipAddress;
+        this.homePageUrl = homePageUrl!=null?homePageUrl.intern():null;
```

Comment:
Looking at the discovery metadata for my apps these urls are absolute and contain the hostname. Would there actually be many copies of the same value?